### PR TITLE
Added requirements and gap report

### PR DESCRIPTION
**Pull request by carioca-au**
- organised
- added gap report
- updated requirements.txt

### CODE REVIEW

File: dags/sentinel_2/sentinel_2_gap_report.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    # Store report in the S3 bucket
+    # s3_report = S3(conn_id=CONN_SENTINEL_2_SYNC)
+    #
+    # s3_report.put_object(
+    #     bucket_name=SENTINEL_2_SYNC_BUCKET_NAME,
+    #     key=key,
+    #     region=REGION,
+    #     body="\n".join(missing_scenes),
+    # )
```
Comment (alexgleith) [inline dags/sentinel_2/sentinel_2_gap_report.py]: Why aren't we writing the report any more?

File: dags/sentinel_2/sentinel_2_gap_report.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    logging.info(
+        f"missing_scenes {missing_scenes if len(missing_scenes) < 100 else list(missing_scenes)[0:2]}"
+    )
+    logging.info(f"10 first missing_scenes {list(missing_scenes)[0:10]}")
+    print(f"Wrote inventory to: s3://{SENTINEL_2_SYNC_BUCKET_NAME}/{key}")
```
Comment (alexgleith) [inline dags/sentinel_2/sentinel_2_gap_report.py:145]: If you didn't write the results, don't say you did

File: dags/sentinel_2/sentinel_2_gap_report.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    if len(orphaned_keys) > 0:
+        output_filename = datetime.today().isoformat() + "_orphaned.txt"
+        key = REPORTING_PREFIX + output_filename
+        # s3_report.put_object(
+        #     bucket_name=SENTINEL_2_SYNC_BUCKET_NAME,
+        #     key=key,
+        #     region=REGION,
+        #     body="\n".join(orphaned_keys),
+        # )
```
Comment (alexgleith) [inline dags/sentinel_2/sentinel_2_gap_report.py]: ??

File: requirements.txt [hunk 4]
```diff
@@ -1,10 +1,9 @@
...
-SQLAlchemy == 1.3.23
+SQLAlchemy==1.3.23
 
 # Used for Landsat data sync
-rasterio~=1.1.8
-stactools-cli==0.1.4
-stactools-core==0.1.4
-stactools-landsat==0.1.4
-pystac==0.5.4
+# On Windows install GDAL (GDAL‑3.1.4‑cp36‑cp36m‑win_amd64.whl) and Rasterio (rasterio‑1.1.8‑cp36‑cp36m‑win_amd64.whl) from https://www.lfd.uci.edu/~gohlke/pythonlibs/
```
Comment (alexgleith) [inline requirements.txt:4]: ???

File: requirements.txt [hunk 5]
```diff
@@ -1,10 +1,9 @@
...
+stactools-cli==0.1.6
+stactools-core==0.1.6
+stactools-landsat==0.1.6
+pystac==0.5.6
```
Comment (alexgleith) [inline requirements.txt:8]: Do we need these?

File: dags/sentinel_2/sentinel_2_gap_report.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+)
+from utils.inventory import InventoryUtils
+from utils.sync_utils import read_csv_from_gzip
+
+default_args = {
+    "owner": "Airflow",
+    "start_date": datetime(2020, 7, 24),
+    "email": ["systems@digitalearthafrica.org"],
+    "email_on_failure": True,
+    "email_on_success": True,
```
Comment (alexgleith) [inline dags/sentinel_2/sentinel_2_gap_report.py:35]: Please don't email on success

File: dags/sentinel_2/sentinel_2_gap_report.py [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+    message = (
+        f"{len(missing_scenes)} scenes are missing from "
+        f"s3://{SENTINEL_2_SYNC_BUCKET_NAME} and {len(orphaned_keys)} "
+        f"scenes no longer exist in s3://sentinel-cogs"
+    )
+    print(message)
+
+    if len(missing_scenes) > 200 or len(orphaned_keys) > 200:
+        raise AirflowException(message)
```
Comment (alexgleith) [inline dags/sentinel_2/sentinel_2_gap_report.py:169]: Let's do the same with Landsat :-)

**Review by alexgleith — CHANGES_REQUESTED** [review]
I don't get why you've turned off report writing.

File: dags/sentinel_2/sentinel_2_gap_report.py (hunk 1 above)
Comment (carioca-au) [reply dags/sentinel_2/sentinel_2_gap_report.py]: You had an old version I already uncommented that

File: dags/sentinel_2/sentinel_2_gap_report.py (hunk 2 above)
Comment (carioca-au) [reply dags/sentinel_2/sentinel_2_gap_report.py:147]: I'm writing. I wasn't writing when it was in DEV

File: dags/sentinel_2/sentinel_2_gap_report.py (hunk 3 above)
Comment (carioca-au) [reply dags/sentinel_2/sentinel_2_gap_report.py]: outdated

File: requirements.txt (hunk 5 above)
Comment (carioca-au) [reply requirements.txt:8]: Just if we restart doing the transfer through here, but once they were already in the requirements I just updated them.
I can remove that, but the logic script is still under utils

File: dags/sentinel_2/sentinel_2_gap_report.py (hunk 6 above)
Comment (carioca-au) [reply dags/sentinel_2/sentinel_2_gap_report.py:38]: All good

File: dags/sentinel_2/sentinel_2_gap_report.py (hunk 7 above)
Comment (carioca-au) [reply dags/sentinel_2/sentinel_2_gap_report.py:171]: sure

**Review by alexgleith — APPROVED** [review]

**Pull request merged**